File: src/main/java/github/rainbowmori/ofro/Ofro.java
package github.rainbowmori.ofro;

import org.bukkit.plugin.java.JavaPlugin;

import github.rainbowmori.ofro.config.ConfigManager;
import github.rainbowmori.ofro.listeners.MVTeleportEvents;

public class Ofro extends JavaPlugin {

  private static ConfigManager configManager;

  public static ConfigManager getConfigService() {
    return configManager;
  }

  public static void reloadConfigManager() {
    configManager.save();
    configManager = new ConfigManager();
  }

  public static Ofro getPlugin() {
    return getPlugin(Ofro.class);
  }

  @Override
  public void onEnable() {
    getServer().getPluginManager().registerEvents(new MVTeleportEvents(), this);
  }
}

File: src/main/java/github/rainbowmori/ofro/Ofro.kt
package github.rainbowmori.ofro

import github.rainbowmori.ofro.config.ConfigManager
import github.rainbowmori.ofro.listeners.MVTeleportEvents
import org.bukkit.plugin.java.JavaPlugin

class Ofro : JavaPlugin() {
  override fun onEnable() {
    configManager = ConfigManager()
    saveResource("README.txt", true)
    server.pluginManager.registerEvents(MVTeleportEvents(),this)
  }

  override fun onDisable() {}

  companion object {
    @JvmStatic
    lateinit var configManager: ConfigManager
      private set

    fun reloadConfigManager() {
      configManager.save()
      configManager = ConfigManager()
    }

    val plugin: Ofro
      get() = getPlugin(Ofro::class.java)
  }
}
